ray-operator/controllers/ray/batchscheduler/yunikorn/yunikorn_scheduler.go

```diff
@@ -0,0 +1,77 @@
+package yunikorn
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	schedulerinterface "github.com/ray-project/kuberay/ray-operator/controllers/ray/batchscheduler/interface"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	SchedulerName                    string = "yunikorn"
+	PodApplicationIDLabelName               = "applicationId"
+	PodQueueLabelName                       = "queue"
+	RayClusterApplicationIDLabelName        = "yunikorn.apache.org/application-id"
+	RayClusterQueueLabelName                = "yunikorn.apache.org/queue-name"
+)
+
+type YuniKornScheduler struct {
+	log logr.Logger
+}
+
+type YuniKornSchedulerFactory struct{}
+
+func GetPluginName() string {
+	return SchedulerName
+}
+
+func (y *YuniKornScheduler) Name() string {
+	return SchedulerName
+}
+
+func (y *YuniKornScheduler) DoBatchSchedulingOnSubmission(ctx context.Context, app *rayv1.RayCluster) error {
+	// yunikorn doesn't require any resources to be created upfront
+	// this is a no-opt for this implementation
+	return nil
+}
+
+func (y *YuniKornScheduler) populatePodLabels(app *rayv1.RayCluster, pod *v1.Pod, sourceKey string, targetKey string) {
+	// check labels
```

Comment:
Is it necessary to enable users to configure both `labels` and `annotations`? Maybe `annotations` are enough.

Reply:
Actually, we can do labels only but not the annotations. Pls see doc here: https://yunikorn.apache.org/docs/user_guide/workloads/workload_overview.

Reply:
This is probably relevant here: https://issues.apache.org/jira/plugins/servlet/mobile#issue/YUNIKORN-1351